lib/pf/Report.pm

```diff
@@ -37,15 +38,13 @@ has 'person_fields', (is => 'rw', isa => 'ArrayRef[Str]');
 
 has 'node_fields', (is => 'rw', isa => 'ArrayRef[Str]');
 
-# empty since no queries are prepared upfront
-sub Report_db_prepare {}
-
 sub generate_sql_query {
     my ($self, %infos) = @_;
     my $logger = get_logger;
 
     my $sqla = SQL::Abstract::More->new();
     my $and = [];
+    $infos{search}{type} //= $self->base_conditions_operator;
```

Comment:
Any specific reason for adding this? Doesn't seem related to DAL work

Reply:
Yes there a undef bug when I was refactoring.
I will added add that to devel.